### DISCUSSION TITLE
Information required

### DISCUSSION THREAD

**Discussion by ChristophCaina** [PiotrMachowski/Home-Assistant-custom-components-Saver#8] [Q&A]
Hi, sorry - I don't know, where I should put this questions instead... therefore, sorry for using this section...

I am running some custom sensors which calculate measurements from my solaredge integration:
- monthly Energy Production per kWp
- yearly Energy Production per kWp

as well as some other statistics.

I would like to know, if this integration is able to store these values (preferred when the timeframe for their measurement ends, end of the month, end of the year...)
And if the states could be saved in the database...

In this case, I would be able to create a statistic with SQL Sensors for example and use the states to compare the production from last month with the current month, etc.

**Comment by PiotrMachowski** [comment]
Hi!
Sorry, I forgot to enable "Discussions".

I'm not sure this is a correct integration for this usage... Have you tried [Utility Meter](https://www.home-assistant.io/integrations/utility_meter/)?

**Comment by ChristophCaina** [comment]
Hi, thanks for your reply.

Yes - the utility_meter was what I've tried at first - but with with this, the current value of the sensor will be ignored and counting does start at 0.0 as soon as the utility meter was created.

Finally, I've saw some issues with the sum of the sensor... I would have expected (as far as I could get from other sensors), that the sum will always be the difference between current state and last state - but in my case, it was always the same as the state...

statistics
![grafik](https://user-images.githubusercontent.com/26391061/139105779-818ca4df-a68c-481b-9003-c2245f07d3c9.png)

statistics_short_term
![grafik](https://user-images.githubusercontent.com/26391061/139105915-b44ae07c-97c0-44db-8916-76ad71343a6f.png)

if you do have an idea why this could be - and how I could get the curren sensor value as state in the database, that would also be an option :)
